src/core/service.c

```diff
@@ -1426,6 +1426,26 @@ static int service_spawn(
         if (r < 0)
                 return r;
 
+        if (IN_SET(s->kill_context.kill_mode, KILL_MIXED, KILL_CONTROL_GROUP) && !s->kill_context.send_sigkill) {
+                _cleanup_free_ char *text = NULL;
+                _cleanup_free_ char *f = NULL;
+                r = cg_get_path(SYSTEMD_CGROUP_CONTROLLER, UNIT(s)->cgroup_path, "cgroup.procs", &f);
+                if (r < 0)
+                        return r;
+                if (f) {
+                        r = read_one_line_file(f, &text);
+                        if (r < 0)
+                                return r;
+                        /* text not empty, process must exist */
+                        if (*text != '\0') {
+                                r = -EINVAL;
+                                log_unit_error_errno(UNIT(s), r, "Will not start SendSIGKILL=no service of type KillMode=cgroup or mixed while process %s is still active", text);
```

Comment:
hmm, could you please add a comment here, explaining why the combination of these settings means it's probably safe to refuse starting in this case.